Forms/FormBuilder.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace SilverStripe\Forms;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * A service which can generate a form for a given record and controller
+ */
+interface FormBuilder {
+
+	/**
+	 * @return DataObject
+	 */
+	public function getRecord();
+
+	/**
+	 * @return Controller
+	 */
+	public function getController();
```

Comment:
Do you need a controller reference if you want to implement a generic form builder for a record? It's not available in `getCMSFields()` unless you use `Controller::curr()`. The form building might rely on controller context though (request vars, HTTP headers), so there will be form builders which require a reference. Hmmm...